### ISSUE TITLE
[BUG] Request Params {mergeParams: true}

### ISSUE THREAD

**Issue by gelito** [nestjs/nest#263]
## I'm submitting a...

<pre><code>
[ ] Regression 
[X] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

I have a controller that checks a nested URL like `@Controller('cats/:cat/dogs')` 

Now, is impossible to access to this `:cat` parameter inside the controller methods using the `@Param()` or `@Req()` decorator. 

## Expected behavior

The ability of access to all parameters of the URL, not only internals parameter

## Minimal reproduction of the problem with instructions

```
@Controller('cats/:cat/dogs')
export class DogsController {

  
  @Get(':dog')
  async findAll(@Req() Request, @Param('dog') dog , @Param('cat') cat ) {
    console.log('Request.params', Request.params);
    console.log('dog', dog);
    console.log('cat', cat);
    return [];
  }
}
```

## Changing proposal

Seeing the code the more easy way to change it is to add, on the express-adapter.js, on line 9
```
return express.Router();
```

The `{mergeParams: true}` flag of the [Router](http://expressjs.com/es/api.html#express.router)
```
return express.Router({mergeParams: true});
```
that preserve the `req.params` values from the parent router. If the parent and the child have conflicting param names, the child’s value take precedence.

## Environment

<pre><code>
Nest version: 4.3.1

 
For Tooling issues:
- Node version: v8.9.0  
- Platform: Linux  

Others:

</code></pre>

**Comment by wbhob** [comment]
I believe Nest has just one global Router instance. Correct me if I'm wrong, Kamil.

The point of this being that there is one instance of shared parameters, rather than having to worry about merging them; that is, since there is one router instance, they are merged by default.

**Comment by gelito** [comment]
I think so, but, if you reproduce the code, you will see that `console.log('cat', cat);`return `cat undefined`

**Comment by kamilmysliwiec** [comment]
Wow, such a shame. I was convinced that this option is here already. I'm sorry, will be available in the incoming release! thanks @gelito

**Comment by kamilmysliwiec** [comment]
It's available since `4.4.0` 🙂
